File: Kotlin-Spring/src/main/java/org/codecop/lcdnumbers/DigitFactoryImpl.java
package org.codecop.lcdnumbers;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

/**
 * Creates (LCD) digits from patterns.
 */
@Service
public class DigitFactoryImpl implements DigitFactory {

    @Autowired
    private Patterns patterns;

    @Override
    public Digit create(int digit) {
        return new Digit(digit, patterns);
    }
}

File: Kotlin-Spring/src/main/kotlin/org/codecop/lcdnumbers/DigitFactoryImpl.kt
package org.codecop.lcdnumbers

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

/**
 * Creates (LCD) digits from patterns.
 */
@Service
class DigitFactoryImpl : DigitFactory {
    @Autowired
    private lateinit var patterns: Patterns
    
    override fun create(digit: Int) = Digit(digit, patterns)
}